custom_components/tahoma/binary_sensor.py

```diff
@@ -73,6 +73,12 @@
         device_class=binary_sensor.DEVICE_CLASS_VIBRATION,
         is_on=lambda state: state == STATE_DETECTED,
     ),
+    OverkizBinarySensorDescription(
```

Comment:
That’s much "complex" than this. There is 4 available values: 

```JSON
{
      "type": "DiscreteState",
      "values": [
        "dead",
        "lowBattery",
        "maintenanceRequired",
        "noDefect"
      ],
      "qualifiedName": "core:SensorDefectState"
    },
```

See what does the homebridge integration: https://github.com/dubocr/homebridge-tahoma/blob/97bd290ff9cb466ca6095c694518e9a931b93f60/src/mappers/RainSensor.ts#L24

Reply:
Not sure how to achieve this in our case... Where it is split between binary sensor and sensor.

Reply:
I think we really need to create a BatterySensor class to handle the logic. We can detect battery device though an attribute `core:PowerSourceType`:

```
"attributes": [
                {
                    "name": "core:MinSensedValue",
                    "type": 1,
                    "value": 50
                },
                {
                    "name": "core:MaxSensedValue",
                    "type": 1,
                    "value": 100000
                },
                {
                    "name": "core:MeasuredValueType",
                    "type": 3,
                    "value": "core:LuminanceInLux"
                },
                {
                    "name": "core:Manufacturer",
                    "type": 3,
                    "value": "Somfy"
                },
                {
                    "name": "core:PowerSourceType",
                    "type": 3,
                    "value": "battery"
                }
            ],
```

Reply:
Fine by me, but will still be pretty complex if we also want to handle situations like `maintenanceRequired`.

Reply:
As did homebridge, we can map maintenanceRequired to dead.

Reply:
> As did homebridge, we can map maintenanceRequired to dead.

But currently we cannot create a binary_sensor and sensor for the same state key.

Reply:
So we just have to create a sensor and not a binary sensor.